headerbp/headerbp.go

```diff
@@ -319,14 +360,36 @@ func WithHeaderSetter(setter func(key, value string)) SetOutgoingHeadersOption {
 	}
 }
 
+type setOutgoingIdempotencyKey struct{}
+
+func HasSetOutgoingHeaders(ctx context.Context, options ...HasSetOutgoingHeadersOption) bool {
+	cfg := &hasSetOutgoingHeaders{}
+	WithHasSetOutgoingHeadersOptions(options...).ApplyToHasSetOutgoingHeaders(cfg)
+	hasSet, ok := ctx.Value(setOutgoingIdempotencyKey{}).(bool)
+	hasSet = hasSet && ok
+	if hasSet {
```

Comment:
nit: as long as you are using two receiver type casting (which avoids panicking if the type mismatches), you don't have to check `ok`, as when `ok` is false (the type is not a bool), `hasSet` will always be false (the zero value of bool). so this can be simplified into:
```suggestion
	if hasSet, _ := ctx.Value(setOutgoingIdempotencyKey{}).(bool); hasSet {
```